component/src/main/java/org/wso2/extension/siddhi/io/http/sink/util/HttpSinkUtil.java

```diff
@@ -128,42 +131,102 @@ public String trustStorePassword(ConfigReader sinkConfigReader) {
      * @param clientStorePass      the client trust store pass path.
      * @return set of sender configurations.
      */
-    public Set<SenderConfiguration> getSenderConfigurations(Map<String, String> httpStaticProperties, String
-            clientStoreFile, String clientStorePass) {
-        Set<SenderConfiguration> senderConf;
+    public static SenderConfiguration getSenderConfigurations(Map<String, String> httpStaticProperties, String
+            clientStoreFile, String clientStorePass, ConfigReader configReader) {
+        SenderConfiguration httpSender = new SenderConfiguration(httpStaticProperties
+                .get(HttpConstants.PORT));
         if (httpStaticProperties.get(HttpConstants.SCHEME).equals(HttpConstants.SCHEME_HTTPS)) {
-            SenderConfiguration httpsSender = new SenderConfiguration(httpStaticProperties
-                    .get(HttpConstants.PORT));
-            httpsSender.setTrustStoreFile(clientStoreFile);
-            httpsSender.setTrustStorePass(clientStorePass);
-            httpsSender.setId(httpStaticProperties.get(HttpConstants.TO));
-            httpsSender.setScheme(httpStaticProperties.get(HttpConstants.SCHEME));
-            senderConf = new HashSet<>(Collections.singletonList(httpsSender));
+            httpSender.setTrustStoreFile(clientStoreFile);
+            httpSender.setTrustStorePass(clientStorePass);
+            httpSender.setId(httpStaticProperties.get(HttpConstants.TO));
+            httpSender.setScheme(httpStaticProperties.get(HttpConstants.SCHEME));
+            //populateSenderStaticConfiguration(httpSender, configReader);
```

Comment:
Same problem here :) Why have we commented out some of the lines?

Reply:
Fixed in dd6916be405bc6fc81078deffdfe7b687a1bf283